src/domain/flow-system/services/src/flow/flow_executor_impl.rs

```diff
@@ -523,17 +525,26 @@ impl MessageConsumerT<TaskProgressMessage> for FlowExecutorImpl {
                     let mut flow = Flow::load(flow_id, flow_event_store.as_ref())
                         .await
                         .int_err()?;
-                    flow.on_task_running(message.event_time, message.task_id)
-                        .int_err()?;
-                    flow.save(flow_event_store.as_ref()).await.int_err()?;
+                    if flow.status() != FlowStatus::Finished {
```

Comment:
If we abort the flow will it go into `Finished` state before task is cancelled? I thought abort just sets the flag but we will wait for inner tasks to finish (successfully or be cancelled) before the flow is marked `Finished`. If latter this check seems weird.

Reply:
If we abort the flow, it will go to `Finished` state itself, and will try to cancel tasks in the queue.
If a task is already running, it will finish, but we will ignore this event from the flow point of view.